magda-io/magda
# Issue/1458 Fixed DAP connector Extra / & Some improvement on CSW connector

### What this PR does

Fixes #1458 
- Changes to CSW connector so it will exit 1 when remote endpoint return non-valid data
  - Currently, the system will report unhandled errors
- Fixed Format minions cut mime type incorrectly
  - This will fix DAP connector  (and possibly other data source) Extra `/` in format issue

### Test
Tested on Dev cluster `issue-1458`.

As we didn't touch dap connector, no DAP record aspects will not be updated if we re-crawl DAP (unless DAP updated their data). Thus, re-crawl won't trigger `format minion`.

Also, tried to send a request to `format minion` `re-crawl` API to trigger the re-crawl on the whole database --- this just took too long to go through 1m events--- I have to break it in the half way and try something else.

So the recommended way to test it is:
- Search `records` table for any records with id contains keyword `csiro`
- Note down the `sourceTag`
- Then, search `records` table for any records with that `sourceTag`
- Order by `lastupdate` ASC and note down `lastupdate` of the first record
- Change format-minion lasteventid to this value in database
- Restart `format-minion`

After that, you can go to UI to confirm the result:

![image](https://user-images.githubusercontent.com/674387/44898505-76f4c680-ad42-11e8-8adc-aa5e0f19e14b.png)

### Deploy

The easiest way will probably be changing format-minion lasteventid to one --- just in case other data sources also have datasets with mime type in its format field. 

### Checklist

-   [x] I've assigned this PR to someone (if you don't know who to assign it to, pick @AlexGilleran)
